src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -195,7 +197,7 @@ private IServiceCollection BuildCommonServices(out AggregateException hostingSta
                 var exceptions = new List<Exception>();
 
                 // Execute the hosting startup assemblies
-                foreach (var assemblyName in _options.HostingStartupAssemblies)
+                foreach (var assemblyName in _options.HostingStartupAssemblies.Distinct(StringComparer.OrdinalIgnoreCase))
```

Comment:
Hence why I added it as an exception instead of a warning before. Not sure if it is worth it here just for this message.